pontoon/base/tests/test_models.py

```diff
@@ -0,0 +1,167 @@
+from datetime import datetime
+
+from django_nose.tools import assert_equal, assert_true
+from django.test.utils import override_settings
+
+from pontoon.base.models import User
+from pontoon.base.tests import (
+    IdenticalTranslationFactory,
+    TranslationFactory,
+    TestCase,
+    UserFactory)
+
+
+class UserTranslationManagerTests(TestCase):
+    @override_settings(EXCLUDE=('excluded@example.com',))
+    def test_excluded_contributors(self):
+        """
+        Checks if contributors with mails in settings.EXCLUDE are excluded
+        from top contributors list.
+        """
+        included_contributor = TranslationFactory.create(user__email='included@example.com').user
+        excluded_contributor = TranslationFactory.create(user__email='excluded@example.com').user
+
+        top_contributors = User.translators.with_translation_counts()
+        assert_true(included_contributor in top_contributors)
+        assert_true(excluded_contributor not in top_contributors)
+
+    def test_users_without_translations(self):
+        """
+        Checks if user contributors without translations are returned.
+        """
+        active_contributor = TranslationFactory.create(user__email='active@example.com').user
+        inactive_contributor = UserFactory.create(email='inactive@example.com')
+
+        top_contributors = User.translators.with_translation_counts()
+        assert_true(active_contributor in top_contributors)
+        assert_true(inactive_contributor not in top_contributors)
+
+    def test_unique_translations(self):
+        """
+        Checks if contributors with identical translations are returned.
+            """
+
+        unique_translator = TranslationFactory.create().user
+        identical_translator = IdenticalTranslationFactory.create().user
+        top_contributors = User.translators.with_translation_counts()
+
+        assert_true(unique_translator in top_contributors)
+        assert_true(identical_translator not in top_contributors)
+
+
+    def test_contributors_order(self):
+        """
+        Checks if users are ordered by count of contributions.
+        """
+        users = UserFactory.create_batch(5)
+
+        for i, user in enumerate(users):
+            TranslationFactory.create_batch(5-i, user=user)
+
+        top_translations_totals = [user.translations_count for user in User.translators.with_translation_counts()]
+        assert_equal(top_translations_totals, [5, 4, 3, 2, 1])
```

Comment:
You'll want to check the actual user objects rather than their counts to ensure that the exact users we expect are ordered.